### DISCUSSION TITLE
OpenSSL's QUIC Implementation Supports RFC 9221

### DISCUSSION THREAD

**Discussion by voluntas** [openssl/openssl#22539] [Q&A] [answered]
Are there any plans for OpenSSL's QUIC implementation to support the Unreliable Datagram Extension (
rfc9221)?

I am very much looking forward to OpenSSL's QUIC implementation.

I couldn't find the task on GitHub Projects or the roadmap, so I asked here.

Sorry if this is not the right place to ask.

**Comment by mattcaswell** [comment, marked as answer]
Yes, its on our radar and we would like to implement it. However, once 3.2 is finally released our next priority will be server side support.

I just created a project issue for it so we keep it in mind during our planning:

https://github.com/openssl/project/issues/266

**Reply by voluntas** [reply]
Thanks for the quick response.

I am going to use OpenSSL's QUIC as the basis for Media over QUIC, so I am very glad that RFC 9221 is going to be supported in OpenSSl for QUIC.

**Reply by hlandau** [reply]
Once server support is delivered (which we hope to deliver in 3.3 next year) delivery of the datagram extension, as the most popular QUIC extension, should be a key priority. Whether it makes 3.3 is a question of how quickly we are able to complete feature development on server support, but I would like to see it happen sooner rather than later, as I appreciate the significant use of it and the relative difficulty of implementing it should not be too high.

**Reply by voluntas** [reply]
Thanks for the detailed description! I am very excited to hear that it is a priority.

And I missed this document here, which mentions support for the QUIC datagram extension.
https://github.com/openssl/openssl/blob/master/README-QUIC.md
